From eb4d23989f5eef995c6d16d48c69f948f970ae4d Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 18 Nov 2020 12:19:39 +0200
Subject: [PATCH] [SW-12754] gaudi: security: prot pcie drain regs

The PCIe HBW/LBW_DRAIN_CFG/TIMEOUT are not used by the host driver and
should be protected as well.
Protect the PCIe HBW and LBW drain related registers.

Change-Id: I8671691673ac7a205e284c3d657ea0104852877b
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 506a6df..fe84029 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -764,17 +764,7 @@
 		PROT_BIT(mmPCIE_WRAP_RR_ELBI_RD_SEC_REG_CTRL));
 	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
-	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x280 - 0xD00 */
-	set_prot_bits_range_all(prot_base, SNGL_REG_PROT_SIZE * 21);
-
-	prot_base += SNGL_REG_PROT_SIZE * 21;	/* offsets: 0xD00 - 0xD80 */
-	mask = ~(PROT_BIT(mmPCIE_WRAP_HBW_DRAIN_TIMEOUT) &
-		PROT_BIT(mmPCIE_WRAP_HBW_DRAIN_CFG) &
-		PROT_BIT(mmPCIE_WRAP_LBW_DRAIN_TIMEOUT) &
-		PROT_BIT(mmPCIE_WRAP_LBW_DRAIN_CFG));
-	regmap_writel(mask, PRIV_PROT_REG(prot_base));
-
-	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0xD80 - 0xF80 */
+	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x280 - 0xF80 */
 	set_prot_bits_range_all(prot_base, PROT_SIZE_MAX - (prot_base - base));
 }
 
